common/interpolation/include/interpolation/interpolation_utils.hpp

```diff
@@ -15,48 +15,78 @@
 #ifndef INTERPOLATION__INTERPOLATION_UTILS_HPP_
 #define INTERPOLATION__INTERPOLATION_UTILS_HPP_
 
+#include <algorithm>
 #include <array>
+#include <functional>
+#include <limits>
 #include <stdexcept>
 #include <vector>
 
 namespace interpolation_utils
 {
+constexpr double EPS = std::numeric_limits<double>::epsilon();
+
+template <class T, typename std::enable_if_t<std::is_floating_point_v<T>, bool> * = nullptr>
+bool isEqual(T a, T b)
+{
+  return std::abs(a - b) < std::numeric_limits<T>::epsilon();
+}
+
+// Strictly monotonic increasing
 inline bool isIncreasing(const std::vector<double> & x)
 {
   if (x.empty()) {
-    throw std::invalid_argument("Points is empty.");
+    throw std::invalid_argument("The vector x is empty.");
   }
 
-  for (size_t i = 0; i < x.size() - 1; ++i) {
-    if (x.at(i) >= x.at(i + 1)) {
-      return false;
-    }
-  }
+  // x(i) <= x(i+1) condition check
+  const auto & is_increasing_it = std::adjacent_find(x.begin(), x.end(), std::greater_equal<>());
```

Comment:
Any reason to add reference here?

Reply:
In addition, how about
```
const bool is_increasing = std::adjacent_find(x.begin(), x.end(), std::greater_equal<>()) == x.cend();
return is_increasing;
```
or
```
return std::adjacent_find(x.begin(), x.end(), std::greater_equal<>()) == x.cend();
```
since `is_increasing_it` does not make sense to me.

Ignore my comment if you don't think it's better.

Reply:
Yes you are right. I put _it formiterator.

Reply:
Changed to clear naming.